src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -75,16 +76,18 @@ void buildScatterSelector(
         if (inserted)
         {
             if (max_parts && partitions_count >= max_parts)
```

Comment:
> we always increment it by one as far as I see, so shouldn't `partitions_count` always be equal to `max_parts` in case of this error?

Thanks @nickitat , you're correct.  I misread the iteration above. I will update this as well as fix the bug above.